weave/integrations/anthropic/anthropic_sdk.py

```diff
@@ -1,89 +1,83 @@
 import importlib
 from collections.abc import AsyncIterator, Iterator
 from functools import wraps
-from typing import (
-    TYPE_CHECKING,
-    Any,
-    Callable,
-    Optional,
-    Union,
-)
+from typing import Any, Callable
 
 import weave
-from weave.trace.op_extensions.accumulator import _IteratorWrapper, add_accumulator
+from weave.trace.op import AsyncIterableContext, SyncIterableContext
 from weave.trace.patcher import MultiPatcher, SymbolPatcher
+from weave.trace.weave_client import Call
+
+
+def should_accumulate(call: Call) -> bool:
+    return bool(call.inputs.get("stream"))
 
-if TYPE_CHECKING:
-    from anthropic.lib.streaming import MessageStream
-    from anthropic.types import Message, MessageStreamEvent
-
-
-def anthropic_accumulator(
-    acc: Optional["Message"],
-    value: "MessageStreamEvent",
-) -> "Message":
-    from anthropic.types import (
-        ContentBlockDeltaEvent,
-        Message,
-        MessageDeltaEvent,
-        TextBlock,
-        Usage,
-    )
-
-    if acc is None:
-        if hasattr(value, "message"):
-            acc = Message(
+
+class AnthropicCallback:
+    def before_yield(self, call: Call, value: Any) -> None:
+        from anthropic.types import (
+            ContentBlockDeltaEvent,
+            Message,
+            MessageDeltaEvent,
+            TextBlock,
+            Usage,
+        )
+
+        if call.output is None:
```

Comment:
Accumulation happens directly on the `call.output` itself.  Since each `Call` is unique to the execution, this is safe.

One nice side effect of accumulating directly on `call` is that it's a more faithful representation of what the user sees.  If you stop the stream midway, the the output will look like a partial output